mazerunner/src/main/java/com/bugsnag/android/mazerunner/scenarios/MultiClientApiKeyScenario.kt

```diff
@@ -0,0 +1,23 @@
+package com.bugsnag.android.mazerunner.scenarios
+
+import android.content.Context
+import com.bugsnag.android.Bugsnag
+import com.bugsnag.android.Client
+import com.bugsnag.android.Configuration
+import com.bugsnag.android.disableAllDelivery
+
+/**
+ * Configures two Bugsnag clients with different API keys. The error should be captured by each,
+ * and the correct API key should be used for both.
+ */
+internal class MultiClientApiKeyScenario(config: Configuration,
+                                         context: Context) : MultiClientScenario(config, context) {
```

Comment:
Some of my initial confusion was from this scenario being a subclass, unlike the others, and that two `Scenario` classes are combined into a single test case. While this approach reuses code, it obscures the intent as thus far a `Scenario` class corresponded one-to-one to a `Scenario` in a feature. It would be semantically clearer to have a single `Scenario` class which does different things in the first run vs a later run, perhaps by setting additional extended data on the intent.

Rough example where Scenarios can load `intent.getStringExtra` via `getRuntimeVariable`:

```kotlin
    override fun run() {
        // configure two clients
        firstClient = Client(context, config)
 
        Thread.sleep(10) // enforce request order
        val secondConfig = Configuration("abc123")
        secondConfig.endpoint = config.endpoint
        secondConfig.sessionEndpoint = config.sessionEndpoint
        secondClient = Client(context, secondConfig)

        // cause an exception while offline
        if getRuntimeVariable("OfflineException") == "on" {
            disableAllDelivery(firstClient!!)
            disableAllDelivery(secondClient!!)
            throw IllegalArgumentException("MultiClientApiKeyScenario")
        }
    }
```